semgrep/semgrep/core_runner.py

```diff
@@ -35,7 +35,11 @@
 
 
 def get_re_matches(patterns_re: List[Tuple], path: Path) -> List[PatternMatch]:
-    contents = path.read_text()
+    try:
+        contents = path.read_text()
+    except UnicodeDecodeError:
```

Comment:
Do we want to be more general than `UnicodeDecodeError` here, e.g. just catch `Exception` and warn the user something went wrong? I took a look at the `read_text` docs: https://docs.python.org/3/library/pathlib.html?highlight=path#pathlib.Path.read_text and it looks like the `errors` kwarg can be utilized too. It looks like `errors` defaults to `strict` (makes sense here, we can't easily regex non-text contents), which [will raise a `ValueError`](https://docs.python.org/3/library/functions.html#open) if there's an encoding error. So here we got `ValueError` -> [`UnicodeError`](https://docs.python.org/3/library/exceptions.html#UnicodeError) -> `UnicodeDecodeError`.

Generally, though, it's better to fail quick and loud and learn about cases we didn't consider. I'd say we should catch `UnicodeDecodeError` here, QA this functionality over a large corpus, look for failures, and patch this up with our learnings. Thoughts?

Reply:
I'd leave it like this before #785 is done, and like you said, find more bugs to fix that way that would get their handling filed as new issues.